hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/streams/ContractSelfAdminKeyValidationOp.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.spec.utilops.streams;
+
+import static com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.freezeOnly;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sleepFor;
+import static com.hedera.services.bdd.spec.utilops.streams.StreamValidationOp.readMaybeRecordStreamDataFor;
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+
+import com.hedera.services.bdd.junit.support.validators.ContractSelfAdminKeyValidator;
+import com.hedera.services.bdd.spec.HapiSpec;
+import com.hedera.services.bdd.spec.utilops.UtilOp;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * A {@link UtilOp} that validates that the selected nodes' application or platform log contains or
+ * does not contain a given pattern.
+ */
+public class ContractSelfAdminKeyValidationOp extends UtilOp {
+    private static final long BUFFER_MS = 500L;
+    private final long newContractNum;
+
+    public ContractSelfAdminKeyValidationOp(final long newContractNum) {
+        this.newContractNum = newContractNum;
+    }
+
+    @Override
+    protected boolean submitOp(@NonNull final HapiSpec spec) throws Throwable {
+        // Freeze the network
+        allRunFor(
```

Comment:
The contract admin key validator only applies in certain contract create cases and the should not be used to validate all contract deployments.  This is why I chose to not include it in the `RECORD_STREAM_VALIDATORS` list

Reply:
Ah!  Well, maybe `StreamValidationOp` should be generalized for more cases since it seems to try to handle the flakiness itself.  Anyway, see #16559 - and maybe add some suggestions there?